wire-golden-files/src/main/kotlin/squareup/wire/mutable/MutableHeader.kt

```diff
@@ -0,0 +1,112 @@
+// Code generated by Wire protocol buffer compiler, do not edit.
+// Source: squareup.wire.mutable.Header in squareup/wire/mutable_types.proto
+@file:Suppress(
+  "DEPRECATION",
+  "RUNTIME_ANNOTATION_NOT_SUPPORTED",
+)
+
+package squareup.wire.mutable
+
+import com.squareup.wire.FieldEncoding
+import com.squareup.wire.Message
+import com.squareup.wire.ProtoAdapter
+import com.squareup.wire.ProtoReader
+import com.squareup.wire.ProtoWriter
+import com.squareup.wire.ReverseProtoWriter
+import com.squareup.wire.Syntax.PROTO_2
+import com.squareup.wire.WireField
+import com.squareup.wire.`internal`.JvmField
+import kotlin.Any
+import kotlin.Boolean
+import kotlin.Deprecated
+import kotlin.DeprecationLevel
+import kotlin.Int
+import kotlin.Long
+import kotlin.Nothing
+import kotlin.String
+import kotlin.Suppress
+import kotlin.UnsupportedOperationException
+import okio.ByteString
+
+public class MutableHeader(
+  @field:WireField(
+    tag = 1,
+    adapter = "com.squareup.wire.ProtoAdapter#UINT64",
+    schemaIndex = 0,
+  )
+  public var id: Long? = null,
+  override var unknownFields: ByteString = ByteString.EMPTY,
+) : Message<MutableHeader, Nothing>(ADAPTER, unknownFields) {
+  @Deprecated(
+    message = "Shouldn't be used in Kotlin",
+    level = DeprecationLevel.HIDDEN,
+  )
+  override fun newBuilder(): Nothing = throw UnsupportedOperationException("newBuilder() is unsupported for mutable message types")
+
+  override fun equals(other: Any?): Boolean {
```

Comment:
Did you mean that we should only support referential equality ? 

I think it's also okay for us to also compare the values. Otherwise it might just end up being confusing.